wagtail/wagtailimages/views/images.py

```diff
@@ -228,9 +229,18 @@ def delete(request, image_id):
         return permission_denied(request)
 
     if request.method == 'POST':
-        image.delete()
-        messages.success(request, _("Image '{0}' deleted.").format(image.title))
-        return redirect('wagtailimages:index')
+        try:
+            image.delete()
+            messages.success(request, _("Image '{0}' deleted.").format(image.title))
+            return redirect('wagtailimages:index')
+        except ProtectedError as e:
+            item = e.protected_objects[0]
+            messages.warning(request, _(
+                "Image '{0}' can't be deleted because is in use in '{1}'."
+            ).format(image.title, item.title), buttons=[
+                messages.button(reverse('wagtailadmin_pages:edit', args=(item.id,)),
```

Comment:
Same kind of problem here. This assumes that the item is a page. If it was a snippet, then the URL would be wrong. If it was not editable in the wagtail admin, there is no way of making a nice button for it at all.